src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,216 @@
+        Car car = new Car(resultSet.getString("model"), manufacturer);
+        car.setId(carId);
+        return car;
+    }
+
+    private List<Driver> getDriverListByCarId(Long carId) {
+        String query = "SELECT d.id, d.name, d.license_number "
+                + "FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "JOIN drivers d ON cd.driver_id = d.id "
+                + "WHERE c.id = ? AND c.deleted = FALSE;";
+        List<Driver> drivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getDriversByCarStatement
+                        = connection.prepareStatement(query)) {
+            getDriversByCarStatement.setLong(1, carId);
+            ResultSet resultSet = getDriversByCarStatement.executeQuery();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject("id", Long.class);
+                String name = resultSet.getString("name");
+                String licenseNumber = resultSet.getString("license_number");
+                Driver driver = new Driver(name, licenseNumber);
+                driver.setId(id);
+                drivers.add(driver);
+            }
+            return drivers;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get list of drivers by car id" + carId,
+                    throwable);
+        }
+    }
+
+    private boolean insertDrivers(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUE (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertDriversStatement
+                        = connection.prepareStatement(query)) {
+            insertDriversStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertDriversStatement.setLong(2, driver.getId());
+                insertDriversStatement.executeUpdate();
+            }
+            return insertDriversStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't add drivers to car " + car, throwable);
+        }
+    }
+
+    private void removeDrivers(Car car) {
+        String deleteDriversQuery = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateCarStatement =
+                         connection.prepareStatement(deleteDriversQuery)) {
+            updateCarStatement.setLong(1, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete drivers from car id "
+                    + car.getId(), throwable);
+        }
+    }
+
+    private void setDrivers(List<Car> carList) {
+        if (carList != null) {
```

Comment:
it's probably redundant, because we always initializing  List<Car> cars = new ArrayList<>() before method call